wps-cloudformation-template.yaml

```diff
@@ -1102,52 +1102,60 @@ Resources:
   BatchJobLogsLambdaPermission:
     Type: AWS::Lambda::Permission
-    Condition: IsProdEnvironment
+    Condition: "ConfigureSumoLogic"
     Properties:
       Action: lambda:invokeFunction
       FunctionName: !GetAtt SumoLogicLambdaFunction.Arn
       Principal: !Sub 'logs.${AWS::Region}.amazonaws.com'
       SourceArn: !Join ["", ["arn:aws:logs:", !Ref "AWS::Region", ":", !Ref "AWS::AccountId", ":log-group:/aws/batch/job:*"]]
   BatchJobLogsSubscriptionFilter:
     Type: "AWS::Logs::SubscriptionFilter"
-    Condition: IsProdEnvironment
+    Condition: "ConfigureSumoLogic"
     DependsOn: BatchJobLogsLambdaPermission
     Properties:
       DestinationArn: !GetAtt SumoLogicLambdaFunction.Arn
       FilterPattern: "" # Will need to update the filter pattern or create a log stream. Currently all the batch job logs will be fetched.
       LogGroupName: "/aws/batch/job"
   RequestHandlerLogsLambdaPermission:
     Type: AWS::Lambda::Permission
-    Condition: IsProdEnvironment
+    Condition: "ConfigureSumoLogic"
     Properties:
       Action: lambda:invokeFunction
       FunctionName: !GetAtt SumoLogicLambdaFunction.Arn
       Principal: !Sub 'logs.${AWS::Region}.amazonaws.com'
-      SourceArn: !Join ["", ["arn:aws:logs:", !Ref "AWS::Region", ":", !Ref "AWS::AccountId", ":log-group:/aws/lambda/", !Ref RequestHandlerLambdaFunction,":*"]]
+      SourceArn: !Join ["", ["arn:aws:logs:", !Ref "AWS::Region", ":", !Ref "AWS::AccountId", ":log-group:", !Ref RequestHandlerLogGroup,":*"]]
   RequestHandlerLambdaLogsSubscriptionFilter:
     Type: "AWS::Logs::SubscriptionFilter"
-    Condition: IsProdEnvironment
+    Condition: "ConfigureSumoLogic"
     DependsOn: RequestHandlerLogsLambdaPermission
     Properties:
       DestinationArn: !GetAtt SumoLogicLambdaFunction.Arn
       FilterPattern: ""
-      LogGroupName: !Join ["", ["/aws/lambda/", !Ref RequestHandlerLambdaFunction]]
+      LogGroupName: !Ref RequestHandlerLogGroup
   JobStatusLogsLambdaPermission:
     Type: AWS::Lambda::Permission
-    Condition: IsProdEnvironment
+    Condition: "ConfigureSumoLogic"
     Properties:
       Action: lambda:invokeFunction
       FunctionName: !GetAtt SumoLogicLambdaFunction.Arn
       Principal: !Sub 'logs.${AWS::Region}.amazonaws.com'
-      SourceArn: !Join ["", ["arn:aws:logs:", !Ref "AWS::Region", ":", !Ref "AWS::AccountId", ":log-group:/aws/lambda/", !Ref JobStatusLambdaFunction,":*"]]
+      SourceArn: !Join ["", ["arn:aws:logs:", !Ref "AWS::Region", ":", !Ref "AWS::AccountId", ":log-group:", !Ref JobStatusLogGroup,":*"]]
   JobStatusLambdaLogsSubscriptionFilter:
     Type: "AWS::Logs::SubscriptionFilter"
-    Condition: IsProdEnvironment
+    Condition: "ConfigureSumoLogic"
     DependsOn: JobStatusLogsLambdaPermission
     Properties:
       DestinationArn: !GetAtt SumoLogicLambdaFunction.Arn
       FilterPattern: ""
+      LogGroupName: !Ref JobStatusLogGroup
+  JobStatusLogGroup:
```

Comment:
Do these really need their own log group? It creates an enormous amount of groups in cloudwatch... can a single stack log everything to the same group?

Reply:
> Do these really need their own log group? It creates an enormous amount of groups in cloudwatch... 



These log groups will be created automatically if we don't specify it in the cf after the first run of the lambda. The problem is that some resources i.e. JobStatusLogsLambdaPermission uses these log groups while provisioning and fails to provision as these log groups have not been created yet. 

> can a single stack log everything to the same group?



Stack will still log everything to the default group. The way around can be to stream the logs from all the different log groups to a single group. I think we don't need to do it as we are already streaming all these logs to sumo.

Reply:
Ok, as discussed I think we will need to revisit this later... it's horribly messy having all of these unmanaged log groups with unlimited retention being created every time a stack is deployed.

So as I understand it, the solution would be for each lambda function, get the stack to pre-create the log group with the desired config (i.e. retention).

Reply:
Correct... You may also want to look into the log group for other resources like API Gateway to set the retention period. Also, we can have a separate job (for entire account) which sets the retention period or delete the old logs.